core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushdownLimitIntoWindow.java

```diff
@@ -79,6 +82,15 @@ public Result apply(LimitNode node, Captures captures, Context context)
     {
         WindowNode source = captures.get(childCapture);
 
+        if (node.isOrderSensitive()) {
+            // Do not pushdown Limit with input ordering if ordering scheme is not satisfied by source
+            List<LocalProperty<Symbol>> desiredProperties = node.getInputOrdering().get().toLocalProperties();
+            if (LocalProperties.match(source.getOrderingScheme().get().toLocalProperties(), desiredProperties).stream()
+                    .anyMatch(Optional::isPresent)) {
+                return Result.empty();
+            }
```

Comment:
I would suggest we don't do this as part of this PR. Given that we're relying on LocalProperties, which, as I mentioned elsewhere, are not currently designed for use anywhere other than in AddExchanges/AddLocalExchanges, it will require more careful thought.

I would also change `getOrderingScheme()` from tracking the actual ordering to just tracking the fields it depends on.